File: 第二天/UIImageView的frame设置/UIImageView的frame设置/ViewController.m
//
//  ViewController.m
//  UIImageView的frame设置
//
//  Created by ityike on 2016/12/7.
//  Copyright © 2016年 ityike. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    
    // 设置frame的方式
    
    // 方式一：
//    UIImageView *imageView = [[UIImageView alloc] init];
//    imageView.image = [UIImage imageNamed:@"1"];
//    imageView.frame = CGRectMake(100, 100, 300, 682);
    
    //or
//    imageView.frame = (CGRect){{100, 100},{300,682}};
    
    // 方式二：
//    UIImageView *imageView = [[UIImageView alloc] init];
//    // 创建一个UIImage对象
//    UIImage *image = [UIImage imageNamed:@"2"];
//    // 设置frame
//    imageView.frame = CGRectMake(0, 0, image.size.width, image.size.height);
//    // 设置图片
//    imageView.image = image;
    
    // 方式三：
//    UIImage *image = [UIImage imageNamed:@"2"];
//    UIImageView *imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, image.size.width, image.size.height)];
//    imageView.image = image;
    
    // 方式四：
    // 创建一个UIImageView对象
    // 注意 initWithImage 默认是有尺寸的-->就是图片的尺寸
    UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"2"]];
    
    // 改变位置
//    imageView.center = CGPointMake(200, 150);
    imageView.center = CGPointMake(self.view.frame.size.width / 2, self.view.frame.size.height / 2);
    
    [self.view addSubview:imageView];
    
}




@end

File: button/button/ViewController.swift
//
//  ViewController.swift
//  button
//
//  Created by ityike on 2016/12/3.
//  Copyright © 2016年 袁 峰. All rights reserved.
//

import UIKit

// @interface ViewController() <UITableViewDatasource>
//class ViewController: UIViewController, UITableViewDataSource {

class ViewController: UICollectionViewController {

    // MARK: - 视图生命周期
    // MARK: 视图加载完成
    override func viewDidLoad() {
        super.viewDidLoad()
/*
        // 创建视图
        let v = UIView(frame: CGRect(x: 20, y: 20, width: 200, height: 150))
        
        // 设置背景颜色
//        v.backgroundColor = UIColor.red
        v.backgroundColor = #colorLiteral(red: 0.7254902124, green: 0.4784313738, blue: 0.09803921729, alpha: 1)        // TODO: 应该修改新的颜色
        
        // 添加到当前视图
        view.addSubview(v)
        
        // 创建图象视图
        let iv = UIImageView(image: #imageLiteral(resourceName: "FullSizeRender"))        // FIXME: 更改头像
        
        iv.center = view.center
        view.addSubview(iv)
 */
        
//        setUI()
    
    }
    /*
    
    // MARK: - UITableViewDataSource
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 20
    }
    
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cellId", for: indexPath)
        // textLabel 是可选的
        cell.textLabel?.text = "\(indexPath.row)"
        return cell
    }
    
    func setUI() {
        // 创建表格
        let tv = UITableView(frame: view.bounds, style: .plain)
        
        // 添加到视图
        view.addSubview(tv)
        
        // 注册可重用 cell
        tv.register(UITableViewCell.self, forCellReuseIdentifier: "cellId")
        
        // 设置数据源
        tv.dataSource = self
    }
    
    */
    
    // MARK: 接受到内存警告
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }


}

